src/main/java/core/Game.java

```diff
@@ -512,14 +520,21 @@ public final AbstractAction oneAction() {
 
         // Get actions for the player
         s = System.nanoTime();
-        List<AbstractAction> observedActions = forwardModel.computeAvailableActions(observation);
+        List<AbstractAction> observedActions = forwardModel.computeAvailableActions(observation, currentPlayer.getParameters().actionSpace);
```

Comment:
I am not keen on the Game having a ForwardModel and using it to get available Actions, and then throwing these away so that the player uses their own Action space. 

As just discussed with Raluca, it would be cleaner if the Player strores the Forward Model / Action Space that it is using. The Game can then use this directly here to give a clean split of responsibility.

The important other thing is that the AbstractGameState does not have a reference (even via Parameters) to the ForwardModel or ActionSpace...these should be Player level parameters.